### ISSUE TITLE
Use redis-resource as Concourse mutex

### ISSUE THREAD

**Issue by gdenn** [starkandwayne/redis-resource#1]
# Feature Request
i am currently checking if i could implement a Mutex for two of my Concourse Pipelines using this redis-resource.

## The Problem:
I got two Pipelines which use the same staging system for bosh deployments but only one of the Pipelines should act at the same time.

## My Solution:
I would create a key staging_lock in the redis service connected by your redis-resource. The value of staging_lock can be true or false.

Now i want my redis-resource in both Pipelines to trigger as soon as the staging_lock value changes to true. One Pipeline starts the work and sets the value of staging_lock to false. After the Pipeline is finished the value is set to true again.

## The Question
Is this currently possible with the redis-resource, or if not would you be interested in having this trigger Feature in you ressource?

best

**Comment by geofffranks** [comment]
Have you seen the pool resource or locker resource? Would either of those fit the bill for you?

https://github.com/concourse/pool-resource

https://github.com/cloudfoundry-community/locker-resource/blob/master/README.md

**Comment by jhunt** [comment]
@dennisjbell thoughts?

**Comment by gdenn** [comment]
Yes i saw the resources, but i don't like deploying another micro service or committing locks to a github repo. 

I would see more value in doing this with your redis resource since it offers us many other advantages. 

Having a trigger on the resource would not only let you implement mutexes in Concourse, but you can use it in many other ways. E.g. if you want to trigger specific jobs of other pipelines. This is very easy with a trigger on a key / value of the redis resource.

You could even implement semaphores with redis triggers, which is great if your system has limited resources but you want to bring at least some parallelism to your pipelines.

**Comment by jhunt** [comment]
I feel like @dennisjbell and I implemented this using the redis-resource, but I can't seem to recall the details.  I'll see if I can ping him via other channels to weigh in here.
